test/utils/wabt_engine.cpp

```diff
@@ -44,6 +57,26 @@ bool WabtEngine::parse(bytes_view input) const
 
 bool WabtEngine::instantiate(bytes_view wasm_binary)
 {
+    wabt::interp::HostModule* hostModule = m_env.AppendHostModule("env");
+    assert(hostModule != nullptr);
+
+    hostModule->AppendFuncExport("adler32", {{wabt::Type::I32, wabt::Type::I32}, {wabt::Type::I32}},
+        [=](const wabt::interp::HostFunc*, const wabt::interp::FuncSignature*,
+            const wabt::interp::TypedValues& args, wabt::interp::TypedValues& results) {
+            const auto offset = args[0].value.i32;
+            const auto length = args[1].value.i32;
+            // TODO: move the memory lookup outside in order to reduce overhead
+            assert(m_env.GetMemoryCount() > 0);
+            auto memory = m_env.GetMemory(0);
+            assert(memory != nullptr);
+            auto memory_data = memory->data;
```

Comment:
what's the type of `memory_data`?

Reply:
it's `vector<char>`

Reply:
Some wabt memory type. Apparently `std::vector<char> data`.